HPCC-23553 Update versioning for spark plugin

Michael-Gardner opened the pull request:
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

Commit:
HPCC-23553 Update versioning for spark plugin

Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

HPCCSmoketest commented:
Process of PR-13893, label: HPCC-23553 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 354c511b9e4738aa76beb35878453b1a985feef9
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-138.ca-central-1.compute.internal

Michael-Gardner commented:
@rpastrana Versioning change for the fetched spark plugin jars.

If we're doing an rc or building off master/x branch, we use wget to hit the sonatype api to pull the SNAPSHOT packages.

Logic for Releases stay relatively the same. Just modified that portion to use the same version variables as the platforms version.cmake file.

Added some error checking after download and ensured the error checks at packaging time still work.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-138.ca-central-1.compute.internal
Sha: 354c511b9e4738aa76beb35878453b1a985feef9
Build: success
Milestone:Install hpccsystems-platform-community_7.8.25-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |877 |877 |0 |310 sec  (00:05:10) |
|test (thor) |815 |815 |0 |743 sec  (00:12:23) |
|test (roxie) |956 |956 |0 |403 sec  (00:06:43) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  702 sec (00:11:42) |  118 sec (00:01:58) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1784 sec (00:29:44) |   19 sec (00:00:19) | 2674 sec (00:44:34) |

rpastrana reviewed (approved):
@Michael-Gardner looks good.

HPCCSmoketest commented:
Process of PR-13893, label: HPCC-23553 starts now.
The reason of this test is: Did not build yet.
Commit ID: 354c511b9e4738aa76beb35878453b1a985feef9
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 354c511b9e4738aa76beb35878453b1a985feef9
Build: success
Milestone:Install hpccsystems-platform-community_7.8.25-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |877 |877 |0 |569 sec  (00:09:29) |
|test (thor) |815 |815 |0 |1013 sec  (00:16:53) |
|test (roxie) |956 |956 |0 |706 sec  (00:11:46) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   16 sec (00:00:16) |  985 sec (00:16:25) |  124 sec (00:02:04) |   22 sec (00:00:22) |   17 sec (00:00:17) | 2646 sec (00:44:06) |   18 sec (00:00:18) | 3828 sec (01:03:48) |
